TypeDB Clients documentation technical review

izmalk opened the pull request:
## DO NOT MERGE this PR after review.

## What is the goal of this PR?

We are rewriting all the docs for TypeDB Clients.

This is a Request for comments, it will be a subject to a series of reviews, so it shouldn't be merged yet, regardless of the review results.

That is a technical review (i.e. subject matter experts fixing and adding content). So the time of our engineers should be spent mostly on:

- Content (correctness, clarity, unambiguousness, completeness)
- Structure (Navigation menu (for the pages on review, see the list above), internal page structure)
- Examples (something does not work or seems like could use some explanation)
- General feeling that something is not right
- Instead of fixing typos and grammar. That being said, all feedback is much appreciated!

Don't feel obliged to fix any errors — just point them out and I'll deal with them or participate in the discussion if I'll need more information to fix.

## What are the changes implemented in this PR?

Full rewrite of TypeDB Clients section.

izmalk committed:
Initial commit

krishnangovindraj commented on 02-clients/02-console.md:
```diff
@@ -0,0 +1,328 @@
+
+## Installation
+
+TypeDB Console distributed with TypeDB server and does not need an installation.
```
You can also get a typed-server without console included.

izmalk replied:
Yeah, you can probably even delete console from the folder, but why would we write about that?
We distribute it (via our Downloads page https://vaticle.com/download#typedb and packages in Homebrew, etc.) with TypeDB Console inside. No need to even mention the possibility to obtain TypeDB without the console IMHO. Or do you think we need it for something?

krishnangovindraj commented on 02-clients/02-console.md:
```diff
@@ -0,0 +1,328 @@
+| `--prefetch`                    | `true|false`   | Enable or disable RPC answer prefetch                          |
+| `--session-idle-timeout`        | `1..[max int]` | Kill idle session timeout (ms)                                 |
+| `--schema-lock-acquire-timeout` | `1..[max int]` | Acquire exclusive schema session timeout (ms)                  |
+| `--read-any-replica`            | `true|false`   | Allow or disallow reads from any replica (TypeDB Cloud only)   |
```
Sometimes we say "TypeDB cloud only", sometimes "TypeDB Cluster only" ?

izmalk replied:
TypeDB Cluster is the old name. I'll change it to the TypeDB Cloud everywhere. It is what we call it for now.

krishnangovindraj commented on 02-clients/02-console.md:
```diff
@@ -0,0 +1,328 @@
+typedb::data::write> insert $p isa person;
+
+{ p iid 0x966e80017fffffffffffffff isa person; }
+typedb::data::write> commit
```
Isn't there usually a `*` in the prompt here to indicate there are changes in the transaction?

izmalk replied:
Good catch. I believe I fixed it. With a major refactoring (renamed and moved a couple of sections to better differentiate between interactive/Non-interactive modes of Console).

krishnangovindraj reviewed (commented):
Reviewed `02-console.md`

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+# TypeDB Studio
+
+TypeDB Studio is an IDE specifically designed to work with 
+[TypeQL]() query language and TypeDB databases.
```
The QL means query language already ;)

izmalk replied:
But the name of the query language is still `TypeQL`, isn't it? Like the name of the most known query language is `SQL`. That is why I suggest that the line is correct (also, missing a link for the term, because we don't have that section yet).

jamesreprise replied:
Could we go with "[...] work with TypeQL (TypeDB's query language) and TypeDB databases."?

izmalk replied:
We can't risk it being "TypeQL and TypeDB databases".

Updated the definition in https://github.com/vaticle/docs/pull/679/commits/8fff3059fd2179df21e5117685910bf9084e84e4
Please check

jamesreprise replied:
agreed, resolved.

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+|      Studio      |      TypeDB      |  TypeDB Cluster  |
+|:----------------:|:----------------:|:----------------:|
+| 2.14.1 to 2.14.2 | 2.14.1 to 2.14.2 |      2.14.1      |
+|      2.11.0      |      2.11.1      | 2.11.1 to 2.11.2 |
```
We can add all of 2.17.0 here!

izmalk replied:
Ideally I would prefer not to mix it up. We have added the version in to the master branch. So if we don't touch the table it should be updated automatically upon merge (not by this PR probably). But that is a big IF, considering many factors. 
Anyway - we can double-check versions in those tables as part of releasing this new docs, just not to update all the PRs all the time new release happens. I think that would be best.

jamesreprise replied:
Makes sense.

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+### Download & install
+
+Download TypeDB distributive/package for any popular OS family from the
+[Download page](https://vaticle.com/download#typedb-studio. 
```
```suggestion
[Download page](https://vaticle.com/download#typedb-studio). 
```

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+toc: false
+---
+
+# TypeDB Studio
```
Would a hero image before everything else be gratuitous? Something like the image from the README:
![image](https://user-images.githubusercontent.com/51956016/235954737-74961ed2-7b23-4891-a693-afcfe3828f02.png)

jamesreprise replied:
TypeDB Studio is inherently a graphical thing, so it makes sense to start with an image. Though, perhaps an image that shows every feature in use could be overwhelming at the beginning of the tutorial.

izmalk replied:
I like the idea in general. Even tho we don't want the picture to be too big. I'll think about a balanced solution here.

izmalk replied:
Fixed. Also the image I added is not as cool because I had to use IAM schema and micro-dataset and they don't have that sweet data clusterization.

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+
+Right after starting TypeDB Studio we can see its main window as it is shown below.
+
+![TypeDB Studio GUI](../images/studio/studio-gui.png)
```
I think this image could use a little work. Some suggestions:
* It would be great if we used our default font (Titillium Web) for the numbers: https://fonts.google.com/specimen/Titillium+Web
* We should try putting boxes around the sections of Studio we're highlighting, preferably with rounded corners.
* Let's use our light green accent (#02DAC9) for the numbers and the boxes rather than the current red.

jamesreprise replied:
Likewise for the rest of the places we've added highlights (arrows, boxes, circles, etc).

izmalk replied:
Implemented with the font, but the Pink color from our new color palette. Haikal chose Pink.

jamesreprise replied:
I love it! Looks much better.

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+#### Run panel
+
+Under the Text editor panel there is the Run panel, but it is collapsed by default. To expand it perform any query or 
+just click on the reversed tick icon in the bottom right corner of TypeDB Studio window, above the cursor location 
```
```suggestion
just click on the upwards-pointing chevron in the bottom right corner of TypeDB Studio window, above the cursor location 
```

jamesreprise replied:
or 'up arrow'.

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+To execute a query written in the Text editor panel we have the **Toolbar** elements located right from the database 
+selection drop-down menu. See below.
+
+![Transaction controls](../images/studio/studio-transaction-controls.png)
```
Let's take this picture again with the database name as 'iam' rather than 'iam-4'.

izmalk replied:
I did that and copy-pasted all buttons in their active (not greyed out) state.

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+  - `read` — to use read transaction
+- [**snapshot**] button — enables snapshot feature that encapsulates transaction in a snapshot of a database's data 
+  until the transaction committed or closed. Write transactions are always snapshoted. Only available for read 
+  transactions.
```
```suggestion
  until the transaction committed or closed. Write transactions are always snapshoted. Always enabled for write transactions, may be disabled for read 
  transactions.
```

izmalk replied:
```suggestion
  until the transaction committed or closed. Write transactions are always snapshoted. May be enabled or disabled for read 
  transactions.
```

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+  transactions.
+- [**infer**] button — enable [inferring data](../0001-typedb/02-dev/06-infer.md) by rules. Only available for read 
+  transactions.
+- [**explain**] button — enable [explanations](../0001-typedb/02-dev/06-infer.md#explain-query) mechanics by providing 
```
\+ "Only available for read transactions."

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+  `explainables` methods for results.
+- [**Transaction status**] (Circle (●) icon) indicator — transaction activity indicator. Green if there is a 
+  transaction 
+  opened by TypeDB Studio at this moment. Grey if there is no opened transaction. Rotating animation — TypeDB 
```
```suggestion
  opened by TypeDB Studio at this moment. Gray if there is no opened transaction. Rotating animation — TypeDB 
```
We have to speak Amer*can now :(

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+
+After selecting the database to work with from a list of all databases on the server Studio connected with, we will 
+see types hierarchy of the selected database in the Types browser (bottom left of the screen). For example, for any 
+empty database (with no schema defined) it will show only the base types:
```
A picture of this would be nice to help the user see they're on the right track.

izmalk replied:
This is not the place to talk about the Types explorer in detail. It's merely a section on how to select a database.

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+
+### Export schema
+
+We can export schema of the database by clicking the [**Export Schema**] button (second in row from left to right) on 
```
The button is labeled "Export Schema Types" to indicate that it does not include rules. The warning below is still valid as it's annoying.

izmalk replied:
Was the third word of the button named added in 2.17? Because in older version of Studio (when the text was written) I do remember it wasn't there.

dmitrii-ubskii commented on 02-clients/00-clients.md:
```diff
@@ -0,0 +1,89 @@
+- Software tools:
+  - [TypeDB Studio](01-studio.md)
+  - [TypeDB Console](02-console.md)
+- [TypeDB drivers](#typedb-drivers) — libraries to connect to a TypeDB server with an application. There are:
```
"TypeDB drivers" — new terminology? This is the first I hear of it.

izmalk replied:
Yes, this is new terminology. And quite consistent one I should say. We can still call everything a TypeDB Client (as long as it connects to a TypeDB server/cluster). There is a picture at the end of the Architecture section.

dmitrii-ubskii commented on 02-clients/00-clients.md:
```diff
@@ -0,0 +1,89 @@
+## Architecture of a TypeDB Client
+
+All TypeDB Clients share a common architecture. A TypeDB Client needs a `gRPC` client library to communicate with 
+the  server. Most languages have `gRPC` libraries.
```
> A TypeDB Client needs a `gRPC` client library to communicate with the  server. Most languages have `gRPC` libraries.

Weird thing to specify.

izmalk replied:
Yeah, it was a leftover from the New Client page. I will delete it.

dmitrii-ubskii commented on 02-clients/00-clients.md:
```diff
@@ -0,0 +1,89 @@
+
+The following schema is the dependency graph of a typical TypeDB Client.
+
+![Structure of a TypeDB Client](../images/client-api/client-structure.png)
```
This diagram is completely arcane. What is `client`? In most current clients (/ drivers?) we have a `core client` and a `cluster client`, `core client` being the common interface.

I don't fully understand whether this is supposed to expose the internal structure, which is more than a little weird on the client intro page, or the user-facing structure, which this diagram is almost entirely orthogonal to. We can discuss and go through this offline.

izmalk replied:
Let's discuss it. As this is auto-generated diagram of modules, that I only redrawn for better perception.

dmitrii-ubskii commented on 02-clients/00-clients.md:
```diff
@@ -0,0 +1,89 @@
+[Quickstart guide](../0001-typedb/01-start/03-quickstart.md).
+
+TypeDB Console is a powerful command line (**CLI**) tool often used to manage TypeDB server and its databases. For more 
+information on how to use it see the [TypeDB Console](02-console.md) page.
```
First, why are we explaining what GUI and CLI here at all?
Second, I'd rather we describe Studio as "a GUI TypeDB client most remarkable for its type explorer and graph visualization." as opposed to it being remarkable for being graphical.

izmalk replied:
1. It is not an explanation. More like decoding the abbreviation. It is a custom to have the abbreviation written with its full version (as well as the abbreviation itself, written in parenthesis) at the first appearance or when it's most convenient. This is the very first appearance, except for the introduction and the list above, where it would be inconvenient.
2. I'm not sure why would we try to shorten it, but I suggest we just switch to its definition given on its own page - it's an IDE.

dmitrii-ubskii commented on 02-clients/00-clients.md:
```diff
@@ -0,0 +1,89 @@
+
+![Structure of a TypeDB Client](../images/client-api/client-structure.png)
+
+Simply put, the main components of any TypeDB Client are the classes 
+and methods to establish a connection to a TypeDB database, execute queries and parse responses. Software tools 
```
"The main components of any database client are the methods that allow us to use the database." This whole paragraph can be cut.

izmalk replied:
1. This is not a paragraph, this is a sentence. Did you mean the whole paragraph or that one sentence?
2. We can delete any part of the docs, but why would we delete that one? It's an important connection between an idea of methods and an idea of an interface to use them.

dmitrii-ubskii commented on 02-clients/06-other-languages.md:
```diff
@@ -5,30 +5,27 @@ longTailKeywords: typedb client, typedb community client, typedb julia client, t
 ---
 
-## TypeDB Clients developed by the community
+# Community TypeDB Client
```
Singular?

izmalk replied:
```suggestion
# Community TypeDB Clients
```

dmitrii-ubskii commented on 02-clients/07-new-client.md:
```diff
@@ -47,30 +51,37 @@ In our case, we only need client-side library compilation. Most languages have P
-![Client Package Structure](../images/client-api/package-structure.png)
+The following diagram shows all the packages (directories) in Client Java and their dependency graph:
+
+![Client Package Structure](../images/client-api/response-structure.png)
```
Not what the diagram shows. Was this meant to stay `package-structure.png`?

dmitrii-ubskii commented on 02-clients/07-new-client.md:
```diff
@@ -47,30 +51,37 @@ In our case, we only need client-side library compilation. Most languages have P
+[Important]
+Import statements are not included in this tutorial, except when importing from external libraries such as the 
+TypeDB protobuf definitions.
+</div>
```
> Import statements are not included in this tutorial

But perhaps they should be?

izmalk replied:
I don't know, you tell me. I would gladly add them if you think they are needed and will provide them.

dmitrii-ubskii commented on 02-clients/node-js/01-node-js-overview.md:
```diff
@@ -0,0 +1,35 @@
+
+# Node.js TypeDB Driver overview
+
+Node.js Driver was developed by Vaticle to enable TypeDB support for Node.js software and developers.
```
Cut this.

dmitrii-ubskii commented on 02-clients/node-js/02-node-js-install.md:
```diff
@@ -0,0 +1,28 @@
+
+## Prerequisites
+
+To use the Node.js Driver, you need a compatible version of TypeDB Server running. Please see the
+[Compatibility Table](01-node-js-overview.md#version-compatibility) to check what version do you need, depending on the TypeDB 
+server version being used.
```
Should the compatibility table be on this page instead? Seems like a natural place for it.

dmitrii-ubskii commented on 02-clients/node-js/03-node-js-tutorial.md:
```diff
@@ -0,0 +1,159 @@
+
+## Importing
+
+In your source, require `typedb-client`.
```
```suggestion
In your source, `require` `typedb-client`.
```

dmitrii-ubskii commented on 02-clients/node-js/03-node-js-tutorial.md:
```diff
@@ -0,0 +1,159 @@
+	client.close();
+}
+
+runBasicQueries("social_network");
```
Does this not need to be `await`ed?

dmitrii-ubskii commented on 02-clients/node-js/04-node-js-api-ref.md:
```diff
@@ -0,0 +1,148 @@
+toc: true
+---
+
+# Node.js API reference
```
Can't help but think that these ought to be separate pages, per class.

dmitrii-ubskii commented on 02-clients/07-new-client.md:
```diff
@@ -47,30 +51,37 @@ In our case, we only need client-side library compilation. Most languages have P
+TypeDB protobuf definitions.
+</div>
 
 <div class="tabs dark">
```
Code indentation is missing in the rendered version.

dmitrii-ubskii commented on 02-clients/07-new-client.md:
```diff
@@ -164,11 +179,17 @@ class TypeDBClient(ABC):
 Create the directory structure: `connection/core` in the root of your project.
-Name the classes depending on language conventions: in Java/TypeScript, `TypeDBClientImpl` and `CoreClient`; in Python, `_TypeDBClient` and `_CoreClient`.
+Name the classes depending on language conventions: in Java/TypeScript, `TypeDBClientImpl` and `CoreClient`; in Python, 
+`_TypeDBClient` and `_CoreClient`.
```
Utterly unnecessary levels of hand-holding which could be better spent on other stuff.

dmitrii-ubskii commented on 02-clients/07-new-client.md:
```diff
@@ -556,34 +582,44 @@ def typedb_client_test():
 
 We recommend using one of our existing Clients as a reference, and copying the implementation into your chosen language.
 
 ## Session and Transaction
```
Half of this tutorial (up to this point) is opinionated walkthrough of basic client (driver) structure that spends most of its non-code text explaining language concepts like naming and making sure to import gRPC, and the rest is a very high-level overview which expects the reader to figure the rest of the owl out.

dmitrii-ubskii commented on 02-clients/07-new-client.md:
```diff
@@ -611,64 +647,75 @@ message Transaction {
+Now, there are two basic patterns to the communications; _single_ responses and _streamed_ responses, both of which 
+are illustrated below.
 
 ![Inside a Transaction Stream](../images/client-api/tx-stream.png)
```
Oh hey. This is where the `response-structure.png` was suppose to go. `tx-stream` is a better name for the diagram though.

izmalk committed:
Fixing old naming TypeDB Cluster into TypeDB Cloud

izmalk committed:
Console major refactoring

izmalk commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,421 @@
+- [**infer**] button — enable [inferring data](../0001-typedb/02-dev/06-infer.md) by rules. Only available for read 
+  transactions.
+- [**explain**] button — enable [explanations](../0001-typedb/02-dev/06-infer.md#explain-query) mechanics by providing 
+  `explainables` methods for results.
```
```suggestion
- [**explain**] button — enable [explanations](../0001-typedb/02-dev/06-infer.md#explain-query) mechanics by providing 
  `explainables` methods for results. Only available for read transactions with **snapshot** and **infer** enabled.
```

izmalk committed:
Apply suggestions for Studio

Co-authored-by: James Williams <james@berserksystems.com>

izmalk committed:
Apply alternative suggestions for Studio

izmalk committed:
Red/green markings screenshot experiment

izmalk committed:
Fix redundant gRPC libraries reference

It doesn't make much sense to have it here. But we do have it in the New client page.

izmalk committed:
Studio screenshots update

izmalk committed:
Studio page minor update

izmalk committed:
Update Studio screenshots (pink)

izmalk committed:
Fix importing in drivers tutorials

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,424 @@
+By double-clicking any type we can open the type browser window that shows all information about the selected type and
+allows us to easily edit the type. 
+
+![Types browser](../images/studio/types-browser.png)
```
This is a type page! The Type Browser is, as above, the panel on which all the types are listed in Studio's main view.

jamesreprise commented on 02-clients/01-studio.md:
```diff
@@ -0,0 +1,424 @@
+
+![Commit the Transaction](../images/studio/project-schema-committed.png)
+
+## Types explorer capabilities
```
We call this the 'Type Browser' in our code, so let's reflect this here too.

izmalk replied:
But why we should stick to the existing terminology if we can improve it? 
Let's discuss what we should call it in our documentation for our users and then adapt the source codes to reflect the naming if needed.

We clearly have these explorer vs browser terms. And we have to name the Types panel (or we can leave it as the Types panel) and the tab with Type detailed information.

**Explorer** is more like exploring all different files/types/objects, moving through the hierarchy. So it's closer to the panel. 

**Browser** is more about seeing a detailed view of a singular type at a time (with the potential ability to move to another). So it's closer to the tab.

That is why I called them this way. We can skip the panel (as we can call it the Types panel) and just call the tab the Type page, the Type tab, or something like that. But that will belittle the features. As we are building an IDE, we should really have some ambition in the naming of functional elements IMHO.

jamesreprise replied:
I see your argument. I think we should create an issue specific to the nomenclature of Studio components and come to agreement on all of the naming that might be more understandable to users if changed.

In the mean time, I think it's important that we don't enter a state where the source code and the documentation diverge in how things are named. It's a tenant of our design philosophy that we have a ubiquitous language (at least per product) that we adhere to across documentation, code and discussions.

izmalk replied:
1. We have discussed the approach with Shane and I think he is more flexible with this. We are re-implementing our external communications framework and rewriting the documentation. It is time to change the terminology now if we think it will help with our external communications. The only question is what will be chosen as the right terminology.
2. There is a meeting about terminology planned. I'm waiting for Shane to approve it before proceeding.

krishnangovindraj commented on 02-clients/java/02-java-install.md:
````diff
@@ -0,0 +1,84 @@
+```
+
+If you want to depend on snapshot versions of Client Java, by referring to the GitHub commit `sha`, you can add our
+snapshot repository to your list of Maven repositories.
````
Do we want the general users to be able to do this?

izmalk replied:
I have no idea. It's what we have in our old docs - https://docs.vaticle.com/docs/client-api/java#add-a-repository-with-typedb-java-client-to-maven

Can you find someone to address your question to and invite him into this discussion?

krishnangovindraj commented on 02-clients/java/02-java-install.md:
```diff
@@ -0,0 +1,84 @@
+## (Optional) Add logging config
+
+By default, the Java Driver uses Logback to print errors and debugging info to standard output. As it is quite verbose, 
+use the following steps to set the minimum log level to `ERROR`:
```
Do you think we should modify the default behaviour? Spamming stdout seems a bit much.

izmalk replied:
I don't have much prior Java experience. Can you address this question to someone more experienced in this matter or invite them in this discussion?

krishnangovindraj commented on 02-clients/java/03-java-tutorial.md:
```diff
@@ -0,0 +1,161 @@
+            // read transaction is open
+            // read transaction must always be closed
+            readTransaction.close();
+        }
```
Why not use the `try (...)` construct for transactions as well?

izmalk replied:
We doing just that down there. I guess the logic was to keep the transaction as a one-liner at this stage. To simplify the code. I think it can help someone to see both variants (with and without the try() block).

izmalk committed:
Update clients structure image

krishnangovindraj commented on 02-clients/java/05-java-query-builder.md:
```diff
@@ -0,0 +1,57 @@
+To use TypeQL, we first add it as a dependency to our `pom.xml`. Don't forget to replace the `{version}` placeholder 
+with exact version of the library.
+
+The latest version of `typeql-lang` can be found in the 
```
> in *vaticle's* public maven repository

might be better? I thought you meant maven central the first time I read this.

izmalk replied:
Fixed this in https://github.com/vaticle/docs/pull/679/commits/1d4086d022b3c4b3c22394a6b65089c29d88072c
As well as added examples for the query builder syntax. Please check them out.

izmalk committed:
Update compatibility tables

izmalk committed:
Apply suggestions for other-clients

izmalk committed:
Update Studio definition

izmalk committed:
New client page update

izmalk committed:
Testing the fix for links test

izmalk committed:
Add examples for Java query builder

And fix the link label for Vaticle's Maven repository

krishnangovindraj commented on 02-clients/references/client.yml:
```diff
@@ -311,50 +301,17 @@ methods:
     java:
       <<: *method-users-delete
-      method: client.users().delete(String username);
+      method: client.users().get(String user).delete();
```
I do not find this in the java client.
The signature I see is the one that's been removed.

izmalk replied:
Fixed by updating API reference to the latest master branch of public docs

krishnangovindraj commented on 02-clients/references/session.yml:
```diff
@@ -103,9 +103,69 @@ methods:
+    common: &method-database-schema
+      title: Retrieve schema of the database in the current session
+      description: |
+        Returns a full schema text as a string.
```
If it's guaranteed to be valid typeql, we could specify since it's useful to know?

krishnangovindraj commented on 02-clients/concept-references/attribute.yml:
```diff
@@ -0,0 +1,203 @@
+        - "`Long`"
+        - "`float`"
+        - boolean
+        - "`Date`"
```
inconsistent formatting

krishnangovindraj commented on 02-clients/concept-references/attribute.yml:
```diff
@@ -0,0 +1,203 @@
+  - method:
+    common: &method-asRemote
+      title: Get a Remote version of the Attribute.
+      description: The remote version uses the given transaction to execute every method call.
```
"execute method calls" is a bit ambiguous. We should probably convey that the methods involve querying the server.

krishnangovindraj commented on 02-clients/concept-references/attribute_type.yml:
```diff
@@ -0,0 +1,247 @@
+  - method:
+    common: &method-getOwners
+      title: Retrieve direct and inherited owners of this Type of Attribute
+      description: Retrieve all Things that own an attribute of this type. Optionally, only fetches Things that own an attribute of this type as a key.
```
`getOwners` retrieves ThingType. We tend to use "Thing" to refer to the instance, rather than the type.

Might be simpler to just say "Retrieves all types that may own an attribute of this type"

krishnangovindraj commented on 02-clients/concept-references/attribute_type.yml:
```diff
@@ -0,0 +1,247 @@
+  - method:
+    common: &method-getOwnersExplicit
+      title: Retrieve direct owners of this Type of Attribute
+      description: Retrieve all Things that directly own an attribute of this type. Optionally, only fetches Things that own an attribute of this type as a key.
```
Is it intuitive enough that "directly owns" means "is explicitly declared to own"

krishnangovindraj reviewed (commented):
pass on concept-references.

izmalk committed:
Fixing API reference files

izmalk committed:
New client page update

izmalk committed:
Client overview pages update

izmalk committed:
Minor updates to Drivers installation
